### PR TITLE
[NEW] Allow request avatar placeholders as PNG or JPG instead of SVG

### LINKED ISSUE #7908: Provide PNG instead of SVG for image caching
Current avatar are sent in either png or svg format, depending on existence of avatar for user.
But image loaders in android or ios doesn't support direct rendering of SVG images. Most of the times, need to write down custom code on top of image loaders to load SVG images. This created problem of inconsistency while loading images ( one avatar is assigned to another user) and caching. I think it's really easy to solve from server side by using simple trick
There are some libraries that allows quick conversion of SVG to PNG specifically for designed platform
1. https://www.npmjs.com/package/svg2png
2. https://www.npmjs.com/package/svg-to-png
3. https://www.npmjs.com/package/save-svg-as-png
You can provide end-point as /avatar/username?format="png";height="20px";width="20px"
which will return png based on required attributes. Those images are easy to parse and in absolute format, so SDK or Android developer don't have to check for SVG or PNG format returned by server.

I hope you guys consider my request :+1:

### PR DESCRIPTION
**Pull request by lindoelio**
@RocketChat/core 

Closes #7908 

Add ability to request the placeholders in PNG or JPG formats passing a query string `format={png,jpg,jpeg}`.

Add ability to request the placeholders in any size smaller than 1024px passing a query string `format=jpg&size=80`, default is 200px

Example https://open.rocket.chat/avatars/@rodrigo.nascimento?format=png&size=500

image

### CODE REVIEW

File: server/startup/avatar.js [hunk 1]
```diff
@@ -86,11 +86,16 @@ Meteor.startup(function() {
 					initials = initials.toUpperCase();
 				}
 
-				const svg = `<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<svg xmlns=\"http://www.w3.org/2000/svg\" pointer-events=\"none\" width=\"50\" height=\"50\">\n<rect height="50" width="50" fill=\"${ color }\"/>\n<text text-anchor=\"middle\" y=\"50%\" x=\"50%\" dy=\"0.36em\" pointer-events=\"auto\" fill=\"#ffffff\" font-family=\"Helvetica, Arial, Lucida Grande, sans-serif\" font-weight="400" font-size="28">\n${ initials }\n</text>\n</svg>`;
-
-				res.write(svg);
-				res.end();
+				const svg = `<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n<svg xmlns=\"http://www.w3.org/2000/svg\" pointer-events=\"none\" width=\"150\" height=\"150\">\n<rect height=\"150\" width=\"150\" fill=\"${ color }\"/>\n<text y=\"50%\" x=\"50%\" pointer-events=\"auto\" fill=\"#ffffff\" font-family=\"Helvetica, Arial, Lucida Grande, sans-serif\" font-weight=\"400\" font-size=\"72\">\n<tspan y=\"67%\" x=\"50%\" text-anchor=\"middle\">${ initials }</tspan>\n</text>\n</svg>`;
```
Comment (lindoelio) [inline server/startup/avatar.js]: Alignment improvements for the generated SVG.

**Comment by karlprieb** [comment]
@rodrigok We want default initials avatar to be PNG by default (if GraphicsMagick or ImageMagick was installed)?

**Comment by lindoelio** [comment]
@rodrigok It's the short time solution about that we did talk in a recent video call so provides a better way for mobile apps to handle user avatar request. Can you make a review?

File: server/startup/avatar.js [hunk 2]
```diff
@@ -79,11 +95,18 @@ Meteor.startup(function() {
 					initials = username.replace(/[^A-Za-z0-9]/g, '').substr(0, 1).toUpperCase();
 				}
 
-				const svg = `<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 80 80\">\n<rect width=\"100%\" height=\"100%\" rx=\"6\" ry=\"6\" fill=\"${ color }\"/>\n<text x=\"50%\" y=\"50%\" dy=\"0.36em\" text-anchor=\"middle\" pointer-events=\"none\" fill=\"#ffffff\" font-family=\"Helvetica, Arial, Lucida Grande, sans-serif\" font-size="50">\n${ initials }\n</text>\n</svg>`;
+				const svg = `<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 80 80\">\n<rect width=\"100%\" height=\"100%\" fill=\"${ color }\"/>\n<text x=\"50%\" y=\"50%\" dy=\"0.36em\" text-anchor=\"middle\" pointer-events=\"none\" fill=\"#ffffff\" font-family=\"Helvetica, Arial, Lucida Grande, sans-serif\" font-size="50">\n${ initials }\n</text>\n</svg>`;
```
Comment (sampaiodiego) [inline server/startup/avatar.js]: when converting to `png` or `jpeg` the result image will have the `viewBox` size. is it 80x80 the size we would like it to have? just for reference, we're using a 120x120 image on profile detail.

Comment (rodrigok) [reply server/startup/avatar.js]: Fixed

**Pull request merged**